### PR TITLE
Add AdminUserDTO and the PublicUserResource

### PR DESCRIPTION
**Pull request by gzsombor**
To separate the publicly accessible user information from the private one.
 Porting https://github.com/jhipster/generator-jhipster/issues/12374

### CODE REVIEW

**Comment by JasonTypesCodes** [comment]
Thanks @gzsombor !  It looks like the gradle-jwt-react sample end to end UI tests need to be checked with these updates.

File: generators/server/templates/src/main/java/package/service/dto/AdminUserDTO.java.ejs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    public AdminUserDTO(User user) {
+        this.id = user.getId();
+        this.login = user.getLogin();
+        this.firstName = user.getFirstName();
+        this.lastName = user.getLastName();
+        this.email = user.getEmail();
+        this.activated = user.getActivated();
+        this.imageUrl = user.getImageUrl();
+        this.langKey = user.getLangKey();
+        //this.createdBy = user.getCreatedBy();
```
Comment (atomfrede) [inline generators/server/templates/src/main/java/package/service/dto/AdminUserDTO.java.ejs:71]: Can this be removed?

**Review by atomfrede — COMMENTED** [review]
Looks good to me. Thanks @gzsombor for contributing this. @JasonTypesCodes We should now decide when to do the last/final release based on 6.x as this feature is already part of 7.x if I am not mistaken (and we will not backport it).

**Comment by gzsombor** [comment]
I think the problem is that this blueprint still use the old - 6.x templates - for the frontend generation. 
I tried to migrate this blueprint to the latest generator, but I didn't have success.

File: generators/server/templates/src/main/java/package/service/dto/AdminUserDTO.java.ejs (hunk 1 above)
Comment (gzsombor) [reply generators/server/templates/src/main/java/package/service/dto/AdminUserDTO.java.ejs:71]: Yes, it was just copy-pasted from the original UserDTO

**Comment by JasonTypesCodes** [comment]
> Looks good to me. Thanks @gzsombor for contributing this. @JasonTypesCodes We should now decide when to do the last/final release based on 6.x as this feature is already part of 7.x if I am not mistaken (and we will not backport it).

I picked up the 1.0 conversation in its issue (#189)

**Comment by gzsombor** [comment]
Sure, I will try to rebase my patch

**Comment by mraible** [comment]
@gzsombor Can you please resolve conflicts or close?

**Comment by gzsombor** [comment]
I've rebased the code, but I can't test it locally, using 'mhipster' or calling 'jhipster --blueprints micronaut' or 'npx jhipster --blueprints micronaut' ends up with this error:

```ERROR! ERROR! Named export 'isReservedTableName' not found. The requested module '../../../jdl/jhipster/reserved-keywords.js' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '../../../jdl/jhipster/reserved-keywords.js';
const { isReservedTableName } = pkg;

file:///mnt/butterfs/projects/jhipster/generator-jhipster/dist/generators/server/support/prepare-entity.mjs:21
import { isReservedTableName } from '../../../jdl/jhipster/reserved-keywords.js';
         ^^^^^^^^^^^^^^^^^^^
SyntaxError: Named export 'isReservedTableName' not found. The requested module '../../../jdl/jhipster/reserved-keywords.js' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '../../../jdl/jhipster/reserved-keywords.js';
const { isReservedTableName } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:190:5)
```

The developer setup is not documented at all, and I can't google it, how to solve. Any recommendation which works with the current JHipster generator?

**Comment by mraible** [comment]
@gzsombor I just merged the [upgrade to JHipster 7.9.3](https://github.com/jhipster/generator-jhipster-micronaut/pull/282), so you might try rebasing again. I created an app last night with the following command after running `npm link`:

```
mhipster jdl ~/dev/generator-jhipster-micronaut/test-integration/samples/gradle-oauth-angular.jdl \
  --skip-jhipster-dependencies --force
```

**Comment by gzsombor** [comment]
Thanks, it turned out, if I don't want to use the development jhipster-generator, it works properly :)

I've rebased, and fixed a couple of smaller issues around the DTO naming, I'm hopeful, that the tests will be green

**Comment by mraible** [comment]
@gzsombor It looks like you need to run `npm run update-snapshot`.

**Comment by gzsombor** [comment]
thanks, I've rebased the change, and fixed the silly search&replace typo bug... This will work now :)

**Comment by gzsombor** [comment]
And everything is green now, we can merge it :)

**Review by mraible — APPROVED** [review]

**Pull request merged**